R/columns.R

```diff
@@ -1,3 +1,78 @@
+#' Set column definitions
+#'
+#' @param widget A [tabulator()] HTML widget.
+#' @param columns The names of the columns the formatter is applied to.
+#' @param title A title that will be displayed for the columns.
+#' @param formatter A formatter definition as generated by a `formatter_*()` function
+#' @param hoz_align (character): The horizontal alignment of the column.
+#' @param ... You can add arguments that will be passed on as updates to the column definition.
+#' Checkout the documentation at \url{https://tabulator.info/docs/}
+#' @returns The updated [tabulator()] HTML widget
+#' @example examples/formatters/formatter_html.R
+#' @export
+set_columns <- function(widget,
+                        columns,
+                        title = NULL,
+                        formatter = NULL,
+                        # TODO editor = NULL,
+                        # TODO validator = NULL,
+                        hoz_align = NULL,
+                        ...){
+  if(is.null(hoz_align) & !is.null(formatter)){
+    hoz_align = switch(formatter$formatter,
+                       html = "left",
+                       plaintext = "left",
+                       textarea = "left",
+                       money = "left",
+                       image = "center",
+                       link = "left",
+                       star = "center",
+                       progress = "left",
+                       tickCross = "center",
+                       datetime = "left",
+                       traffic = "center"
+    )
+  }
+
+  if(!is.null(formatter)){
```

Comment:
Will not work as expected if we want to update multiple columns at once.

Reply:
Does work as expected with multiple columns. Added examples and tests for this case.